tests/fuzz/Makefile

```diff
@@ -0,0 +1,37 @@
+#Modify this to point to the PJSIP location.
+PJBASE=../../
+
+include $(PJBASE)/build.mak
+
+TARGET=Fuzzing
+
+JSON=fuzz-json
+XML=fuzz-xml
+SDP=fuzz-sdp
+STUN=fuzz-stun
+SIP=fuzz-sip
+
+EXTFLAGS=-Wall -Werror
+
+FUZZ_pjlib=$(PJLIB_UTIL_LDLIB) $(PJLIB_LDLIB)
+FUZZ_pjmedia=$(PJMEDIA_CODEC_LDLIB) $(PJMEDIA_VIDEODEV_LDLIB) $(PJMEDIA_AUDIODEV_LDLIB) $(PJMEDIA_LDLIB) $(PJNATH_LDLIB) $(PJLIB_UTIL_LDLIB) $(PJLIB_LDLIB)
+FUZZ_pjnath=$(PJNATH_LDLIB) $(PJLIB_UTIL_LDLIB) $(PJLIB_LDLIB)
+FUZZ_pjsip=$(PJSIP_LDLIB) $(PJSIP_UA_LDLIB) $(PJSIP_SIMPLE_LDLIB)  $(PJMEDIA_CODEC_LDLIB) $(PJMEDIA_VIDEODEV_LDLIB) $(PJMEDIA_LDLIB) $(PJMEDIA_AUDIODEV_LDLIB) $(PJLIB_UTIL_LDLIB) $(PJNATH_LDLIB) $(PJLIB_LDLIB)
+
+all: $(TARGET)
+
+$(TARGET):
+	$(CC) $(PJ_CFLAGS) $(EXTFLAGS) -c $(JSON).c
+	$(CC) $(PJ_CFLAGS) $(EXTFLAGS) -c $(XML).c
+	$(CC) $(PJ_CFLAGS) $(EXTFLAGS) -c $(SDP).c
+	$(CC) $(PJ_CFLAGS) $(EXTFLAGS) -c $(STUN).c
+	$(CC) $(PJ_CFLAGS) $(EXTFLAGS) -c $(SIP).c
+
+	$(CXX) $(PJ_CFLAGS) -o $(JSON) $(JSON).o $(PJ_LDFLAGS) $(LIB_FUZZING_ENGINE) $(FUZZ_pjlib)
```

Comment:
I tested it here and it almost works. It should be:
`$(CXX) $(PJ_CFLAGS) -o $(JSON) $(JSON).o $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)`

Yes, that means that it needs to list all the libraries, instead of select ones, but the compiler should figure it out and link only with the required libraries (in my test here, the resulting binaries have exactly the same size).

The reason we need to use `$(PJ_LDLIBS`) is because it may also contain other libraries that's needed by pjsip libraries (such as certain frameworks).

Reply:
Like This:
```
	$(CXX) $(PJ_CFLAGS) -o $(JSON) $(JSON).o $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)
	$(CXX) $(PJ_CFLAGS) -o $(XML)  $(XML).o  $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)
	$(CXX) $(PJ_CFLAGS) -o $(SDP)  $(SDP).o  $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)
	$(CXX) $(PJ_CFLAGS) -o $(STUN) $(STUN).o $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)
	$(CXX) $(PJ_CFLAGS) -o $(SIP)  $(SIP).o  $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS) $(LIB_FUZZING_ENGINE)
```